src/main/java/seedu/address/logic/parser/EditEventCommandParser.java

```diff
@@ -0,0 +1,82 @@
+import seedu.address.logic.commands.EditEventCommand;
+import seedu.address.logic.commands.EditEventCommand.EditEventDescriptor;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Name;
+
+/**
+ * Parses input arguments and creates a new EditEventCommand object
+ */
+public class EditEventCommandParser implements Parser<EditEventCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the EditEventCommand
+     * and returns an EditEventCommand object for execution.
+     *
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public EditEventCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_DATETIME, PREFIX_DESCRIPTION,
+                        PREFIX_ADD_FRIENDNAME, PREFIX_REMOVE_FRIENDNAME);
+
+        Index index;
+
+        try {
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+        } catch (ParseException pe) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditEventCommand.MESSAGE_USAGE), pe);
+        }
+
+        EditEventDescriptor editEventDescriptor = new EditEventDescriptor();
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            editEventDescriptor.setName(ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get()));
+        }
+        if (argMultimap.getValue(PREFIX_DATETIME).isPresent()) {
+            editEventDescriptor.setDateTime(ParserUtil.parseDateTime(argMultimap.getValue(PREFIX_DATETIME).get()));
+        }
+        if (argMultimap.getValue(PREFIX_DESCRIPTION).isPresent()) {
+            editEventDescriptor.setDescription(ParserUtil.parseDescription(argMultimap.getValue(PREFIX_DESCRIPTION).get()));
+        }
+        parseNamesForEdit(argMultimap.getAllValues(PREFIX_ADD_FRIENDNAME)).ifPresent(editEventDescriptor::setAddFriendNames);
+        parseNamesForEdit(argMultimap.getAllValues(PREFIX_REMOVE_FRIENDNAME)).ifPresent(editEventDescriptor::setRemoveFriendNames);
+
+        if (!editEventDescriptor.isAnyFieldEdited()) {
+            throw new ParseException(EditEventCommand.MESSAGE_NOT_EDITED);
+        }
+
+        return new EditEventCommand(index, editEventDescriptor);
+    }
+
+    /**
+     * Parses {@code Collection<String> names} into a {@code Set<Name>} if {@code names} is non-empty.
+     * If {@code names} contain only one element which is an empty string, it will be parsed into a
+     * {@code Set<Name>} containing zero names.
+     */
+    private Optional<Set<Name>> parseNamesForEdit(Collection<String> names) throws ParseException {
+        assert names != null;
+
+        if (names.isEmpty()) {
+            return Optional.empty();
+        }
+        Collection<String> nameSet = names.size() == 1 && names.contains("") ? Collections.emptySet() : names;
```

Comment:
I am a bit confused, when will it reach the case where Collection<String> nameSet has only one element which is an empty string? maybe I am missing something out but i think`argMultimap.getAllValues` returns an empty list if theres no matching prefixes?

Reply:
Yes, I agree with you it is redundant.